test/test_torch.py

```diff
@@ -1724,6 +1729,17 @@ def test_bool_sub(self):
                                    r"If you are trying to invert a mask, use the `\~` or `bitwise_not\(\)` operator instead.",
                                    lambda: m2 - 1)
 
+    def test_sub(self):
+        for dtype in torch.testing.get_all_dtypes():
+            m1 = torch.tensor([2.34, 4.44], dtype=dtype)
+            m2 = torch.tensor([1.23, 2.33], dtype=dtype)
+
+            if (dtype == torch.half or dtype == torch.bool):
+                self.assertRaises(RuntimeError, lambda: m1 - m2)
+            else:
+                # bfloat16 has a lower precision so we have to have specified precision
+                self.assertEqual(m1 - m2, torch.tensor([1.11, 2.11], dtype=dtype), 0.01)
```

Comment:
umm, isn't this setting lower precision for all dtypes except half and bool?

Reply:
Updated